src/j1cli.js

```diff
@@ -0,0 +1,191 @@
+const J1_API_TOKEN = process.env.J1_API_TOKEN;
+const JupiterOneClient = require('./j1client');
+const { prompt } = require('inquirer');
+const program = require('commander');
+const error = require('./util/error');
+const fs = require('fs');
+
+const EUSAGEERROR = 126;
+
+async function main () {
+  program
+    .version(require('../package').version, '-v, --version')
+    .usage('[options]')
+    .option('-a, --account <name>', 'JupiterOne account name.')
+    .option('-u, --user <email>', 'JupiterOne user email.')
+    .option('-k, --key <apiToken>', 'JupiterOne API access token.')
+    .option('-q, --query <j1ql>', 'Execute a query.')
+    .option('-o, --operation <action>', 'Supported operations: create, update')
+    .option('--entity', 'Specifies entity operations.')
+    .option('--relationship', 'Specifies relationship operations.')
+    .option('--alert', 'Specifies alert rule operations.')
+    .option('-f, --file <dir>', 'Input JSON file.')
+    .parse(process.argv);
+
+  try {
+    const data = await validateInputs();
+    const j1Client = await initializeJ1Client();
+
+    if (program.query) {
+      const res = await j1Client.queryV1(program.query);
+      console.log(JSON.stringify(res, null, 2));
+    }
+    else {
+      switch (program.operation) {
+        case 'create':
+          if (program.entity) {
+            await createEntities(j1Client, Array.isArray(data) ? data : [data]);
+          } else if (program.alert) {
+            await createAlerts(j1Client, Array.isArray(data) ? data : [data]);
+          }
+          break;
+        case 'update':
+          if (program.entity) {
+            await updateEntities(j1Client, Array.isArray(data) ? data : [data]);
+          } else if (program.alert) {
+            await updateAlerts(j1Client, Array.isArray(data) ? data : [data]);
+          }
+          break;
+        default:
+          console.log(`Unsupported operation: ${program.operation}`);
+      }
+    }
+  } catch (err) {
+    error.fatal(`Unexpected error: ${err}`);
+  }
+  console.log('Done!');
+}
+
+// ensure user supplied necessary params
+async function validateInputs () {
+  console.log('Validating inputs...');
+  if (!program.account || program.account === '') {
+    error.fatal('Missing -a|--account flag!', EUSAGEERROR);
+  }
+
+  let data;
+
+  if (!program.query || program.query === '') {
+    if (!program.entity && !program.relationship && !program.alert) {
+      error.fatal('Must specify a query (using -q|--query) or an entity/relationship/rule operation (--entity or --relationship or --alert)', EUSAGEERROR);
+    } else if (!program.operation || program.operation === '') {
+      error.fatal('Must specify command action for entity or rule operation (-c|--command <create|update|delete>)', EUSAGEERROR);
+    } else if (!program.file || program.file === '') {
+      error.fatal('Must specify input JSON file with -f|--file)', EUSAGEERROR);
+    } else if (!fs.existsSync(program.file)) {
```

Comment:
This is already an async function. You may as well `promisify` the `fs.exists` function and just `await` on it.